builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -560,6 +560,18 @@ func (p *provisioner) configureVaultsFunc(gemCmd string, knifeCmd string, confDi
 
 		for vault, items := range p.Vaults {
 			for _, item := range items {
+				
+				removeCmd := fmt.Sprintf("%s vault remove %s %s -C %s -M client %s",
+					knifeCmd,
+					vault,
+					item,
+					p.NodeName,
+					options,
+				)
+				if err := p.runCommand(o, comm, removeCmd); err != nil {
```

Comment:
I confirm, there is no error when client doesn't exist. At least in chef-vault <= 3.2.0.
IMHO chef-vault gem version should be pinned to ensure compatibility.